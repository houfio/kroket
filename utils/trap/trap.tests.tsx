import { renderHook } from '@testing-library/react-hooks';
import * as React from 'react';

import { useTrap } from '.';

it('should return markers when enabled', () => {
  const { result: { current: [begin, end] } } = renderHook(() => useTrap({ current: document.body }, true));

  expect(begin).toMatchSnapshot();
  expect(end).toMatchSnapshot();
});

it('should return fragments when disabled', () => {
  const { result: { current: [begin, end] } } = renderHook(() => useTrap({ current: document.body }, false));

  expect(begin).toMatchSnapshot();
  expect(end).toMatchSnapshot();
});
