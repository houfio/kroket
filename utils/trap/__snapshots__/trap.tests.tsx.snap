// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return fragments when disabled 1`] = `<React.Fragment />`;

exports[`should return fragments when disabled 2`] = `<React.Fragment />`;

exports[`should return markers when enabled 1`] = `
<div
  data-focus-trap=""
  onFocus={[Function]}
  tabIndex={0}
/>
`;

exports[`should return markers when enabled 2`] = `
<div
  data-focus-trap=""
  onFocus={[Function]}
  tabIndex={0}
/>
`;
